### PR TITLE
Livestream tests

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
- [ ] revert actions back to master after actions PR approved
Streaming upload coverage, cases:
- Uploading youtube feed to allocation should work
- upload from feed with delay flag must work
- Upload from feed with a different chunksize must work
- Uploading local webcam feed to allocation should work
- Uploading local webcam feed to allocation with delay specified should work
- Upload local webcam feed with a different chunksize must work
- Uploading youtube feed with negative delay should fail
- Upload from feed with a negative chunksize should fail
- Uploading local webcam feed with negative delay should fail
- Upload from feed with a negative chunksize should fail

FIXMEs found:
- ceil(size/chunksize) is not equalling numblocks when chunksize is user-specified (it should be)
- negative chunksize is not throwing any error but instead starts uploading (it shouldn't)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -110,7 +110,7 @@ jobs:
           blobber_stake_image: latest
 
       - name: "Run System tests"
-        uses: 0chain/actions/run-system-tests@master
+        uses: 0chain/actions/run-system-tests@streamupload
```
Comment (mohsenno1) [inline .github/workflows/ci.yml:113]: Do we need to return it back to 'master'?

**Comment by mohsenno1** [comment]
Great work. Look ok to me.

**Review by mohsenno1 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stewartie4) [reply .github/workflows/ci.yml:113]: good point, is there an actions PR?

**Comment by stewartie4** [comment]
Looks like the tests are hanging until the timeout is reached 
https://0chain.github.io/system_test/livestream-tests/d277cd43b6e8522eaf5cc3bf9b5921891be3b13c/index.html

**Comment by Kishan-Dhakan** [comment]
> Looks like the new tests are hanging until the timeout is reached https://0chain.github.io/system_test/livestream-tests/d277cd43b6e8522eaf5cc3bf9b5921891be3b13c/index.html

Might have been a network issue, I re-ran all jobs and they're passing now

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Kishan-Dhakan) [reply .github/workflows/ci.yml:113]: Yep! Just raised a [PR](https://github.com/0chain/actions/pull/5) on actions repo, once that is merged I'll make a commit to revert this so it gets tested again before merge👍

File: tests/cli_tests/zboxcli_livestream_test.go [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+		localpath := filepath.Join(localfolder, "up.m3u8")
+		err = os.MkdirAll(localpath, os.ModePerm)
+		require.Nil(t, err, "Error in creating the folders", localpath)
+		defer os.RemoveAll(localfolder)
+
+		// Using exec.Command as we need the pid to kill this later
+		cmd := exec.Command("./zbox", "upload", "--allocation", allocationID, "--remotepath", remotepath,
+			"--localpath", localpath, "--feed", feed, "--sync", "--silent", "--wallet", escapedTestName(t)+"_wallet.json",
+			"--configDir", "./config", "--config", configPath)
+		cliutils.Setpgid(cmd)
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_livestream_test.go:56]: why is the command being called in this way? do we need to add retries?

File: tests/cli_tests/zboxcli_livestream_test.go [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+
+		remotepath := "/live/stream.m3u8"
+		localfolder := filepath.Join(os.TempDir(), escapedTestName(t))
+		localpath := filepath.Join(localfolder, "up.m3u8")
+		err = os.MkdirAll(localpath, os.ModePerm)
+		require.Nil(t, err, "Error in creating the folders", localpath)
+		defer os.RemoveAll(localfolder)
+
+		// Using exec.Command as we need the pid to kill this later
+		cmd := exec.Command("./zbox", "upload", "--allocation", allocationID, "--remotepath", remotepath,
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_livestream_test.go:129]: what do you think about extracting  129 - 141 be extracted to a method? i'm in 2 minds

File: tests/cli_tests/zboxcli_livestream_test.go [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+	"testing"
+	"time"
+
+	climodel "github.com/0chain/system_test/internal/cli/model"
+	cliutils "github.com/0chain/system_test/internal/cli/util"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStreamUploadDownload(t *testing.T) {
+	t.Parallel()
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_livestream_test.go:21]: Skip these tests if the youtube-dl dependency is not installed or the livestream is down?

File: tests/cli_tests/zboxcli_livestream_test.go (hunk 2 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_livestream_test.go]: Done!

File: tests/cli_tests/zboxcli_livestream_test.go (hunk 3 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/zboxcli_livestream_test.go]: Yep, refactored this to Start() with and without retries as utils and wrapped upload function around it

**Comment by stewartie4** [comment]
@Kishan-Dhakan like you mentioned, these tests are hanging again - is there any logging we can add to debug why this is happening? We can't merge with this inconsistency as 
1. there may be a code issue
2. we cant have everyone's CI hanging :(

**Comment by Kishan-Dhakan** [comment]
> @Kishan-Dhakan like you mentioned, these tests are hanging again - is there any logging we can add to debug why this is happening? We can't merge with this inconsistency as
> 
>     1. there may be a code issue
> 
>     2. we cant have everyone's CI hanging :(

Yes, definitely we can't merge this in it's current state :( I'm certain this is caused by child processes created by this tests (`ffmpeg`), which is a known [issue in go](https://github.com/golang/go/issues/24050). But I have set a pgid which should technically kill the child processes as well (confirmed on both win and ubuntu local machines thanks to @mohsenno1). I'm trying a few more things, will post an update by today

**Review by stewartie4 — APPROVED** [review]

**Pull request merged**